src/mame/drivers/prodigy.cpp

```diff
@@ -99,6 +120,16 @@ class prodigy_state : public driver_device
 		, m_cb1(*this, "bcd:cb1")
 		, m_cb2(*this, "bcd:cb2")
 		, m_digit(0.0)
+		, m_io_line0(*this, "LINE0")
+		, m_io_line1(*this, "LINE1")
+		, m_io_line2(*this, "LINE2")
+		, m_io_line3(*this, "LINE3")
+		, m_io_line4(*this, "LINE4")
```

Comment:
Is there a reason not to use `required_ioport_array` for this?

Reply:
no idea, will check required_ioport_array out, the refactoring I am planning to do will make it a 2x10 matrix as the keypad is wired that way which I didn't know when starting out.